File: src/test/java/chat/teco/tecochat/chat/domain/chat/QuestionAndAnswerTest.java
package chat.teco.tecochat.chat.domain.chat;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertAll;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator.ReplaceUnderscores;
import org.junit.jupiter.api.Test;

@SuppressWarnings("NonAsciiCharacters")
@DisplayNameGeneration(ReplaceUnderscores.class)
@DisplayName("QuestionAndAnswer 은(는)")
class QuestionAndAnswerTest {

    @Test
    void 복제할_수_있다() {
        // given
        QuestionAndAnswer qna = new QuestionAndAnswer("질문1", "답변1");

        // when
        QuestionAndAnswer copied = qna.copy();

        // then
        assertAll(
                () -> assertThat(copied.question().content()).isEqualTo("질문1"),
                () -> assertThat(copied.answer().content()).isEqualTo("답변1"),
                () -> assertThat(copied).isNotEqualTo(qna)
        );
    }
}

File: src/test/kotlin/chat/teco/tecochat/domain/chat/QuestionAndAnswerTest.kt
package chat.teco.tecochat.domain.chat

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class QuestionAndAnswerTest : StringSpec({

    "질문과 답변을 복사한다" {
        val questionAndAnswer = QuestionAndAnswer("질문", "답변")

        val result = questionAndAnswer.copy()

        result.question.content() shouldBe "질문"
        result.answer.content() shouldBe "답변"
    }
})
